File: leetcode/2232-adding-spaces-to-a-string/solution.go
func addSpaces(s string, spaces []int) string {
    var builder strings.Builder
    builder.Grow(len(s) + len(spaces))
    i := 0
	for _,j := range spaces {
		builder.WriteString(s[i:j])
		builder.WriteByte(' ')
        i = j
	}
    builder.WriteString(s[i:])
    return builder.String()
}

File: leetcode/2232-adding-spaces-to-a-string/solution.kt
class Solution {
    fun addSpaces(s: String, spaces: IntArray): String {
        val builder = StringBuilder(s.length + spaces.size)
        var i = 0
        for(j in spaces) {
            builder.append(s, i, j)
            builder.append(' ')
            i = j
        }
        builder.append(s, i, s.length)
        return builder.toString()
    }
}
